src/python/pants/backend/go/util_rules/build_pkg.py

```diff
@@ -171,6 +179,45 @@ class BuiltGoPackage:
     import_paths_to_pkg_a_files: FrozenDict[str, str]
 
 
+@dataclass
```

Comment:
```suggestion
@dataclass(unsafe_hash=True)
```

Reply:
Do I have to define `__hash__` with `unsafe_hash=True`?

Reply:
No, it tells the dataclass to autogenerate it. It's called `unsafe_hash` because normally it only generates `__hash__` if you have set `frozen=True`. You're saying, "trust me, generate __hash__ even though this type is not immutable"